ts/WoltLabSuite/Core/Component/GoogleMaps/Geocoding.ts

```diff
@@ -115,8 +115,8 @@ class Geocoding {
     this.#element.dataset.googleMapsLat = lat.toString();
     this.#element.dataset.googleMapsLng = lng.toString();
 
-    if (this.#element.hasAttribute("data-google-maps-geocoding-store") && this.#element.form) {
-      this.#store(this.#element.dataset.googleMapsGeocdingStore || "", lat, lng);
+    if ("googleMapsGeocodingStore" in this.#element.dataset && this.#element.form) {
+      this.#store(this.#element.dataset.googleMapsGeocodingStore!, lat, lng);
```

Comment:
Is the exclamation mark actually necessary here? I hoped that TypeScript is able to narrow down the type.

Reply:
My IDE flags this as a possible error, because the value could be undefined.